### PR TITLE
Fixed searchString issue

### PR DESCRIPTION
**Pull request by Ascynx**
searchString should now work like expected.

added a hidden debug boolean which makes searchString spit out the matches used when returning true.

You can enable it using ``/neudiag debug enable search`` and disable it using ``/neudiag debug disable search``

### CODE REVIEW

**Comment by hannibal00212** [comment]
Sorry to see you go, hope you got what you wanted :(

**Comment by Ascynx** [comment]
> Sorry to see you go, hope you got what you wanted :(

so true

**Review by Lulonaut — APPROVED** [review]

**Review by romangraef — CHANGES_REQUESTED** [review]
MFW (my face when) NEUDebugFlag.java

**Comment by Ascynx** [comment]
also something I didn't mention in the commit is about [log4j global logger](https://canary.discord.com/channels/516977525906341928/755816367827058749/1020036793506205726)

File: src/main/java/io/github/moulberry/notenoughupdates/NEUManager.java [hunk 1]
```diff
@@ -432,7 +433,7 @@ public boolean searchString(String toSearch, String query) {
 			}
 			return false;
 		}
-		if (lastStringMatch != -1 && NotEnoughUpdates.INSTANCE.config.hidden.showSearchbarDebug) {
+		if (lastStringMatch != -1 && NotEnoughUpdates.INSTANCE.config.hidden.debugFlags.contains(NEUDebugFlag.SEARCH)) {
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/NEUManager.java:436]: You can just use `NEUDebugFlag.SEARCH.isSet()`

Comment (Ascynx) [reply src/main/java/io/github/moulberry/notenoughupdates/NEUManager.java:436]: how was I supposed to know ;-;

**Review by romangraef — DISMISSED** [review]

**Review by hannibal00212 — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by romangraef — APPROVED** [review]

**Pull request merged**